std/algorithm/comparison.d

```diff
@@ -1565,3 +1568,39 @@ unittest
                 2, "two",
                 ));
 }
+
+/**
+    Returns true if all of the elements in $(D r1) are present in $(D r2).
+    Otherwise, returns false. 
+
+    Params:
+        r1 = Any finite input range
+        r2 = Any finite input range
+
+    Returns:
+        $(D bool)
+*/
+bool isPermutation(Range1, Range2)(Range1 r1, Range2 r2) if (
+    isInputRange!Range1 &&
+    isInputRange!Range2 &&
+    !isInfinite!Range1 && 
+    !isInfinite!Range1
+) {
+    import std.algorithm.searching : countUntil;
+
+    foreach (item; r1) {
+        if (r2.countUntil(item) == -1) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+///
+@safe unittest {
+    assert(isPermutation([1, 2, 3], [3, 2, 1]));
+    assert(isPermutation([1.1, 2.3, 3.5], [2.3, 3.5, 1.1]));
+    assert(isPermutation("abc", "bca"));
+    assert(!isPermutation([1, 2], [3, 4]));
```

Comment:
Need to test with non-array ranges as well. Maybe the stuff in `std.internal.test.dummyrange` may be useful in constructing such test ranges.